File: api/routes/backtest.py
"""策略回测路由"""

from fastapi import APIRouter, HTTPException

from api.models.backtest import (
    BacktestRequest,
    BacktestResponse,
    MultiBacktestRequest,
    PortfolioBacktestRequest,
    PortfolioBacktestResponse,
)
from api.services import backtest_service

router = APIRouter()


@router.post("/shaofu", response_model=BacktestResponse)
def run_shaofu(req: BacktestRequest):
    """少妇战法单股回测"""
    try:
        return backtest_service.run_shaofu(
            ts_code=req.ts_code,
            days=req.days,
            stop_loss_pct=req.stop_loss_pct,
            take_profit_pct=req.take_profit_pct,
            position_pct=req.position_pct,
        )
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"回测失败: {e}")


@router.post("/multi", response_model=BacktestResponse)
def run_multi(req: MultiBacktestRequest):
    """多策略融合回测"""
    try:
        return backtest_service.run_multi(
            ts_code=req.ts_code,
            days=req.days,
            initial_capital=req.initial_capital,
            position_pct=req.position_pct,
            stop_loss_pct=req.stop_loss_pct,
            take_profit_pct=req.take_profit_pct,
        )
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"回测失败: {e}")


@router.post("/portfolio", response_model=PortfolioBacktestResponse)
def run_portfolio(req: PortfolioBacktestRequest):
    """多股票组合回测"""
    try:
        return backtest_service.run_portfolio(
            ts_codes=req.ts_codes,
            days=req.days,
            initial_capital=req.initial_capital,
            position_pct=req.position_pct,
            stop_loss_pct=req.stop_loss_pct,
            take_profit_pct=req.take_profit_pct,
        )
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"回测失败: {e}")

File: frontend/src/api/backtest.ts
import api from './client';
import type { BacktestResult } from './types';

export interface BacktestParams {
  ts_code: string;
  days?: number;
  initial_capital?: number;
  stop_loss_pct?: number;
  take_profit_pct?: number;
  position_pct?: number;
}

export async function runShaofu(params: BacktestParams): Promise<BacktestResult> {
  const { data } = await api.post<BacktestResult>('/backtest/shaofu', params);
  return data;
}

export async function runMulti(params: BacktestParams): Promise<BacktestResult> {
  const { data } = await api.post<BacktestResult>('/backtest/multi', params);
  return data;
}

export async function runPortfolio(tsCodes: string[], params?: Partial<BacktestParams>): Promise<BacktestResult> {
  const { data } = await api.post<BacktestResult>('/backtest/portfolio', {
    ts_codes: tsCodes,
    ...params,
  });
  return data;
}
